api/classes/class-lightweight-api.php

```diff
@@ -170,14 +172,33 @@ public function set_transient( $name, $value ) {
 	}
 
 	/**
+	 * Delete transient.
+	 *
+	 * @param string $name The transient's name.
+	 */
+	public function delete_transient( $name ) {
+		global $wpdb;
+		$name = '_transient_' . $name;
+
+		wp_cache_delete( $name );
+		$wpdb->delete( // phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery
+			$wpdb->options,
+			[ 'option_name' => $name ]
+		);
+
+		$this->debug['delete_query_count'] += 1;
+	}
+
+	/**
+	 * LEGACY FORMAT - one row per reader per prompt.
 	 * Retrieve prompt data.
 	 *
 	 * @param string $client_id Client ID.
 	 * @param string $campaign_id Prompt ID.
 	 * @return object Prompt data.
 	 */
-	public function get_campaign_data( $client_id, $campaign_id ) {
-		$data = $this->get_transient( $this->get_transient_name( $client_id, $campaign_id ) );
+	public function get_campaign_data_legacy( $client_id, $campaign_id ) {
+		$data = $this->get_transient( $this->get_transient_name_legacy( $client_id, $campaign_id ) );
 		return [
```

Comment:
No need to process the return value here - it's processed again by `get_campaign_data`. Just returning `$data` would suffice.

Reply:
Good call, removed in dfc3b19.